### PR TITLE
[llvmonly] Make direct calls to icalls/icall wrappers.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -9072,7 +9072,12 @@ append_mangled_wrapper (GString *s, MonoMethod *method)
 	case MONO_WRAPPER_MANAGED_TO_NATIVE: {
 		append_mangled_wrapper_subtype (s, info->subtype);
 		if (info->subtype == WRAPPER_SUBTYPE_ICALL_WRAPPER) {
-			g_string_append_printf (s, "%s", method->name);
+			const char *name = method->name;
+			const char *prefix = "__icall_wrapper_";
+			if (strstr (name, prefix))
```
Comment (Therzok) [inline mono/mini/aot-compiler.c]:
```suggestion
			if (strstr (name, prefix) == name)
```

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -9072,7 +9072,12 @@ append_mangled_wrapper (GString *s, MonoMethod *method)
 	case MONO_WRAPPER_MANAGED_TO_NATIVE: {
 		append_mangled_wrapper_subtype (s, info->subtype);
 		if (info->subtype == WRAPPER_SUBTYPE_ICALL_WRAPPER) {
-			g_string_append_printf (s, "%s", method->name);
+			const char *name = method->name;
+			const char *prefix = "__icall_wrapper_";
+			if (strstr (name, prefix) == name)
```
Comment (jaykrell) [inline mono/mini/aot-compiler.c]: `strstr` is sloppy and slow here. You basically want a length check and then memcmp, except for maybe a varying number of leading underscores.

File: mono/mini/aot-compiler.c [hunk 3]
```diff
@@ -9072,7 +9072,12 @@ append_mangled_wrapper (GString *s, MonoMethod *method)
...
+				name += strlen (prefix);
+			g_string_append_printf (s, "%s", name);
```
Comment (jaykrell) [inline mono/mini/aot-compiler.c]: `g_string_append`, without the extra `printf` logic?

**Pull request merged**